View3D/Rendering/RenderItems/CorridorRenderItem.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using Microsoft.Xna.Framework.Graphics;
+using System.Linq;
+using SharpDX;
+using View3D.Components.Rendering;
+using Color = Microsoft.Xna.Framework.Color;
+using Matrix = Microsoft.Xna.Framework.Matrix;
+using Vector3 = Microsoft.Xna.Framework.Vector3;
+
+namespace View3D.Rendering.RenderItems
+{
+    public class CorridorRenderItem :IRenderItem
+    {
+        Effect _shader;
+        Vector3 _startPos;
+        Vector3 _endPos;
+        private float _width;
+        Color _colour = Color.Red;
+        private float _halfWidth;
+        private VertexPositionColor[] _startCircleVertecies;
+        private VertexPositionColor[] _endCircleVertecies;
+        private VertexPositionColor[] _connectionCircleVertecies;
+        private VertexPositionColor[] _edgesVertecies;
+
+        public Matrix ModelMatrix { get; set; } = Matrix.Identity;
+
+        public CorridorRenderItem(Effect shader, Vector3 startPos, Vector3 endPos, float width)
+        {
+            _shader = shader;
+            _startPos = startPos;
+            _endPos = endPos;
+            _width = width;
+            Init();
+        }
+
+        public CorridorRenderItem(Effect shader, Vector3 startPos, Vector3 endPos, float width, Color color) : this(shader, startPos, endPos, width)
+        {
+            _colour = color;
+        }
+        
+        private void Init()
+        {
+            _halfWidth = _width / 2;
```

Comment:
if applicable, could this be solved by adding a new function like the void AddCube(Matrix transform, Color color) function?